apps/web-giddh/src/app/new-vs-old-Invoices/new-vs-old-Invoices.component.ts

```diff
@@ -152,11 +152,41 @@ export class NewVsOldInvoicesComponent implements OnInit, OnDestroy {
 
         this.reportYear = this.selectedYear;
 
-        this.store.dispatch(this._NewVsOldInvoicesActions.GetNewVsOldInvoicesRequest(this.NewVsOldInvoicesQueryRequest));
+        this.newVsOldInvoicesService.GetNewVsOldInvoices(this.NewVsOldInvoicesQueryRequest).pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if(response?.status === "success" && response?.body) {
+                this.newVsOldInvoicesData = response?.body;
+
+                this.crdTotal = this.newVsOldInvoicesData.carriedSales?.reduce((p, c) => {
```

Comment:
Consider evaluating `crdTotal`, `invTotal`, `clientTotal` in a single loop, since the source array `newVsOldInvoicesData.carriedSales` is the same